fs/metrics/manager/manager.go

```diff
@@ -0,0 +1,70 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package manager
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/awslabs/soci-snapshotter/fs/metrics/manager/monitor"
+)
+
+var (
+	ErrNoManager        = errors.New("manager not found")
+	ErrNoMonitor        = errors.New("monitor not found")
+	ErrRootManagerUnset = errors.New("root manager is nil")
+)
+
+var gb *globalManager
+
+// G returns the default Global observability manager.
+var G = DefaultGlobalManager
+
+// Manager manages monitors.
+type Manager interface {
+	// RegisterRoot registers the root `Monitor`.
+	RegisterRoot(monitor.Monitor)
+	// Root returns the root `Monitor`.
+	Root() (monitor.Monitor, error)
```

Comment:
Note: Returning an `error` along with the `Monitor`  removes the ability to call a `Monitor` method in a single line:
```go
manager.G().Root().Inc()
```
It does make interacting with monitors less error prone.